tsconfig.base.json

```diff
@@ -9,7 +9,7 @@
     "experimentalDecorators": true,
     "importHelpers": true,
     "target": "es2015",
-    "module": "esnext",
+    "module": "commonjs",
```

Comment:
Unfortunate but I do think this is the best solution for this project. When I moved over to using strings like @ghiobi mentioned I no longer had the issue. The refactoring in this PR probably introduced tighter coupling between entities but makes them more intuitive to work with.

Another benefit is we can use types directly in TypeORM decorators now, making our code easier to read and traverse.